lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -17,14 +31,23 @@ def can_entity_buy_train?(entity)
           def buyable_trains(entity)
             # Cannot buy 2E if one is already owned
             owns_2e = entity.trains.any? { |t| t.name == '2E' }
```

Comment:
These kind of statements make nice micro methods rather than another var.

Down later` if owns_2e?(entity)`